lib/pbench/server/api/__init__.py

```diff
@@ -6,53 +6,43 @@
 
 import os
 import sys
-import hashlib
-from pathlib import Path
-import requests
-import humanize
-import tempfile
 
-from flask import request, jsonify, Flask, make_response
-from flask_restful import Resource, abort, Api
-from werkzeug.utils import secure_filename
+from flask import Flask
+from flask_restful import Api
 
-from pbench.common.logger import get_pbench_logger
 from pbench.server import PbenchServerConfig
 from pbench.common.exceptions import BadConfig
-from pbench.server.utils import filesize_bytes
-
-ALLOWED_EXTENSIONS = {"xz"}
-
-app = None
-
-
-def allowed_file(filename):
-    """Check if the file has the correct extension."""
-    try:
-        fn = filename.rsplit(".", 1)[1].lower()
-    except IndexError:
-        return False
-    allowed = "." in filename and fn in ALLOWED_EXTENSIONS
-    return allowed
+from pbench.server.api.resources.elasticsearch_api import Elasticsearch
+from pbench.server.api.resources.upload_api import Upload, HostInfo
+from pbench.server.api.resources.graphql_api import GraphQL
+from pbench.common.logger import get_pbench_logger
 
 
-def register_endpoints(api, app):
+def register_endpoints(api, app, config):
     """Register flask endpoints with the corresponding resource classes
     to make the APIs active."""
 
     api.add_resource(
-        Upload, f"{app.config['REST_URI']}/upload/ctrl/<string:controller>"
+        Upload,
+        f"{config.rest_uri}/upload/ctrl/<string:controller>",
+        resource_class_args=(config, app.logger),
+    )
+    api.add_resource(
+        HostInfo,
+        f"{config.rest_uri}/host_info",
+        resource_class_args=(config, app.logger),
+    )
+    api.add_resource(
+        Elasticsearch,
+        f"{config.rest_uri}/elasticsearch",
+        resource_class_args=(config, app.logger),
+    )
+    api.add_resource(
+        GraphQL, f"{config.rest_uri}/graphql", resource_class_args=(config, app.logger),
     )
-    api.add_resource(HostInfo, f"{app.config['REST_URI']}/host_info")
-    api.add_resource(Elasticsearch, f"{app.config['REST_URI']}/elasticsearch")
-    api.add_resource(GraphQL, f"{app.config['REST_URI']}/graphql")
-
-
-def create_app():
-    """Create Flask app with defined resource endpoints."""
 
-    global app
 
+def get_server_config():
```

Comment:
Seems like this should probably be in the shell, and simply passed in the config object; no need for shell to import this from the api just to construct the object and pass it back. There's nothing in here that's specific to the API implementation.

Reply:
That's correct but we also need to import this function in `conftest.py` to build the flask client app for pytests and thats why I kept it here. (I can probably put it in the utils file but little skeptical about having to import `PbenchConfigClass` in the utils file coz in the future as we add more stuff and write utility functions we might encounter circular import scenarios. Correct me if I am wrong but util files shouldnt import important classes right?)

Reply:
If we can avoid it, in principle, we don't want to modify the code just to accommodate tests.  Let's have the test environment accommodate the code.